core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1814,6 +1814,12 @@ private[spark] object Utils extends Logging {
       startService: Int => (T, Int),
       conf: SparkConf,
       serviceName: String = ""): (T, Int) = {
+
+    if (startPort != 0 && (startPort < 1024 || startPort >= 65536)) {
```

Comment:
I'd use `require()`; you can write `in [1024,65536)` to express the meaning succinctly; missing space after 0 and I don't think the last parens are needed.